### PR TITLE
HPCC-15987 Minor improvements to disk read processing

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
Jira: https://track.hpccsystems.com/browse/HPCC-15987

**Comment by ghalliday** [comment]
@jakesmith FYI
@mckellyln please review

**Comment by mckellyln** [comment]
Changes look fine to me.

File: rtl/eclrtl/rtlds_imp.hpp [hunk 1]
```diff
@@ -250,17 +250,18 @@ class ECLRTL_API RtlDynamicRowBuilder : implements RtlRowBuilderBase
         self = static_cast<byte *>(_self); 
         maxLength = _maxLength;
     }
-    inline ~RtlDynamicRowBuilder() { clear(); }
+    inline ~RtlDynamicRowBuilder() { if (self) { rowAllocator->releaseRow(self); } }
```
Comment (richardkchapman) [inline rtl/eclrtl/rtlds_imp.hpp]: Minor: odd style to have the extra { } here

**Pull request merged**